src/StakingNode.sol

```diff
@@ -337,21 +362,41 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
         uint256[] memory shares = new uint256[](1);
 
         strategies[0] = beaconChainETHStrategy;
-        shares[0] = sharesAmount;
+        shares[0] = depositSharesAmount;
         // The delegationManager requires the withdrawer == msg.sender (the StakingNode in this case).
         params[0] = IDelegationManagerTypes.QueuedWithdrawalParams({
             strategies: strategies,
             depositShares: shares,
             __deprecated_withdrawer: address(this)
         });
+        uint256 beaconChainSlashingFactor;
```

Comment:
This could be declared inside the if `if (delegatedTo == address(0)) {` as it is only used there